contracts/TiramisuSavingsClub.sol

```diff
@@ -0,0 +1,160 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+
+/// @title Tiramisu Savings Club v1
+/// @author Jerrall, Luis, Mayuko, Prashant
+/// @notice This dapp has a single smart contract architecture. It manages state for all savings clubs in this single contract
+/// @notice v1 was developed during a short period of time (DappCamp) with the goal of being a MVP. See whitepaper for future roadmap ideas
+/// @dev Contraint #1: A user (address) can only belong to a single group
+/// @dev Contraint #2: Only one pending request for money can exist at a time per group
+contract TiramisuSavingsClub {
+    // Store a list of group ids (the indices of this array) and the group owner address (the value at that index)
+    address[] public groupOwners;
+
+    mapping(address => uint) public memberToGroupId; // keeps track of which group each user is in
+    mapping(uint => address[]) public groupsToMembers; // keeps track of which users are in each group
+    mapping(address => string) public memberNames; // store names of users.
+
+    // Index of next member to eligible receive a payout, incremented by 1 after each payout (cycling back to 0)
+    mapping(uint => uint) public nextPayeeIndex;
+    mapping(uint => uint) public groupBalances; // stores the balance of each group
+    
+    mapping(address => uint) deposits; // Deposits per address, per group
+    mapping(address => uint) withdrawals; // Withdrawals per address, per group
+
+    constructor() {
+        // Group id 0 is immediately burned and disallowed for use
+        // Otherwise our mapping lookups will be error prone (can't distinguish default value from the group at index 0)
+        // First real group will be at index 1
+        groupOwners.push(address(0));
+    }
+
+    /// Ensure that the group id is valid
+    /// @param _groupId The index of the group
+    /// @dev Reverts the transaction if the id is invalid
+    /// @dev Group at index 0 is not valid
+    modifier validGroupId(uint _groupId) {
+        require(_groupId >= 0 && _groupId < groupOwners.length, "Inalid group id");
+        _;
+    }
+
+    /// Create a savings group
+    /// @param _members List of addresses that are members of this goup, sorted by payout order
+    /// @param _names List of human readable names that map to addresses - indexed to _members
+    /// @param _ownerIndex Index of the address and name of the owner who is allowed to perform special admin functions
+    /// @return group id
+    function createGroup(address[] memory _members, string[] memory _names, uint _ownerIndex) public returns (uint) {
+        require(_members.length > 0, "Cannot create an empty group");
+        require(_members.length == _names.length, "_members and _names length should match");
+        require(_ownerIndex >= 0 && _ownerIndex < _members.length, "_owner index invalid");
+
+        uint _groupId = groupOwners.length; // future id of this group
```

Comment:
Considering using a [counter](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Counters.sol) to increment and get the next ID instead of relying on the length of the array.